src/format/KdbxReader.cpp

```diff
@@ -92,7 +97,14 @@ Database* KdbxReader::readDatabase(QIODevice* device, QSharedPointer<const Compo
     }
 
     // read payload
-    return readDatabaseImpl(device, headerStream.storedData(), key, keepDatabase);
+    auto* db = readDatabaseImpl(device, headerStream.storedData(), std::move(key), keepDatabase);
```

Comment:
What is the purpose of std::move(...) here? key is a QSharedPointer which does not have a double reference copy constructor (since it's unnecessary).

Reply:
Avoids unneeded reference count manipulations and gets rid of CLion code inspection warnings. 😉

Reply:
CLion is smarter than the Qt docs. QSharedPointer DOES have a move contructor, which just hands over the pointers without incrementing ref counts.